source/base/exceptions.cc

```diff
@@ -336,6 +343,58 @@ void ExceptionBase::generate_message () const
 
 
 
+#ifdef DEAL_II_WITH_MPI
+namespace StandardExceptions
+{
+  ExcMPI::ExcMPI (const int error_code)
+    :
+    error_code (error_code)
+  {}
+
+  void ExcMPI::print_info (std::ostream &out) const
+  {
+    char error_name[MPI_MAX_ERROR_STRING];
+    error_name[0] = '\0';
+    int resulting_length = MPI_MAX_ERROR_STRING;
+
+    // Check the output of the error printing functions. If either MPI function
+    // fails we should just print a less descriptive message.
+    bool error_name_known = true;
+    // get the string name of the error code by first converting it to an error
+    // class.
+    int error_class = 0;
+    int ierr = MPI_Error_class (error_code, &error_class);
+    error_name_known = error_name_known && (ierr == MPI_SUCCESS);
```

Comment:
This condition is funny because you have set `error_name_known` to `true` above :-) I think you can just declare and initialize the variable here with `(ierr == MPI_SUCCESS)`.